### PR TITLE
Disable color testing when color is not supported

### PR DESCRIPTION
**Pull request by bkamins**
Fixes https://github.com/JuliaData/DataFrames.jl/issues/2846

I have tested these changes both with and without color.

### CODE REVIEW

**Comment by bkamins** [comment]
@nalimilan - let us put this on hold, as maybe https://github.com/KristofferC/Crayons.jl/issues/47 will be resolved and then it should not be required.

**Comment by bkamins** [comment]
`Base.get_have_color` is not available in Julia 1.0, but I will keep it as is, since maybe this PR will not be needed given https://github.com/KristofferC/Crayons.jl/issues/47.

**Comment by DilumAluthge** [comment]
In 1.0, instead of `Base.get_have_color()`, I think you can look for the `Base.have_color` book.

I would pursue this as a short-term solution. I'm guessing that https://github.com/KristofferC/Crayons.jl/issues/47 will require some time, and possible some breaking changes.

File: test/io.jl
Comment (bkamins) [inline test/io.jl:633]:
```suggestion
    if VERSION >= v"1.6" && Base.get_have_color()
```

File: test/show.jl
Comment (bkamins) [inline test/show.jl:218]:
```suggestion
    if if VERSION >= v"1.6" && Base.get_have_color()
```

**Comment by bkamins** [comment]
Thank you. I just restricted these tests to Julia 1.6 or newer. I assume we will drop 1.0 support anyway soon when new LTS is out.

@nalimilan - given @DilumAluthge comment I mark it as ready for review.

Thank you!

File: test/show.jl
Comment (bkamins) [inline test/show.jl:219]:
```suggestion
    if VERSION >= v"1.6" && Base.get_have_color()
```

**Review by nalimilan — APPROVED** [review]

**Comment by DilumAluthge** [comment]
Before you merge this, you may want to double-check it fixes the issue by doing e.g.

```
julia --color=no test/runtests.jl
```

Note that you probably don't want to do `Pkg.test`. `Pkg.test` runs the tests in a different process, and I'm not sure if the `--color` flag will get propagated to the subprocess.

**Comment by bkamins** [comment]
> Before you merge this, you may want to double-check that it fixes the issue

As commented in the original post I have done this. Though thank you for double-checking, as these things are indeed tricky (I have run the test again now to be sure 😄).

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**
